src/Corax/Querying/Matches/MultiUnaryMatch.cs

```diff
@@ -552,13 +552,13 @@ public QueryInspectionNode Inspect()
             MultiUnaryItem comparer = _comparers[index];
             var prefix = Constants.QueryInspectionNode.Comparer + index + "_";
             parameters.Add(prefix + Constants.QueryInspectionNode.FieldType, comparer.Type.ToString());
-            parameters.Add(prefix + Constants.QueryInspectionNode.Operation, comparer._leftSideOperation.ToString());
+            parameters.Add(prefix + Constants.QueryInspectionNode.Operation + (comparer._isBetween ? "Left" : string.Empty), comparer._leftSideOperation.ToString());
             parameters.Add(prefix + (comparer._isBetween ? Constants.QueryInspectionNode.Term : Constants.QueryInspectionNode.LowValue), comparer.LeftAsString());
             
             if (comparer._isBetween)
             {
                 parameters.Add(prefix + Constants.QueryInspectionNode.HighValue, comparer.RightAsString());
-                parameters.Add(prefix + Constants.QueryInspectionNode.Operation, comparer._rightSideOperation.ToString());
+                parameters.Add(prefix + Constants.QueryInspectionNode.Operation  + (comparer._isBetween ? "Right" : string.Empty), comparer._rightSideOperation.ToString());
```

Comment:
Those two doesn't seems to be related to the concurrent usage of analyzers issue. Is that the fix for the following exception?

```
from index 'Questions/ByCreationDateAndTags' where Tags = "django" and CreationDate between '2013-04-30T00:00:00.0000000Z' and '2013-05-30T00:00:00.0000000Z' limit 0, 10

System.ArgumentException: An item with the same key has already been added. Key: Comparer0_Operation
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at Corax.Querying.Matches.MultiUnaryMatch`1.Inspect() in C:\Users\omer.ratsaby\Desktop\RDBQA-64_07082024\60\src\Corax\Querying\Matches\MultiUnaryMatch.cs:line 561
   at Raven.Server.Documents.Indexes.Persistence.Corax.CoraxIndexReadOperation.QueryInternal[THighlighting,TQueryFilter,THasProjection,TDistinct](IndexQueryServerSide query, QueryTimingsScope queryTimings, FieldsToFetch fieldsToFetch, Reference`1 totalResults, Reference`1 skippedResults, Reference`1 scannedDocuments, IQueryResultRetriever retriever, DocumentsOperationContext documentsContext, Func`2 getSpatialField, CancellationToken token)+MoveNext()
   at Raven.Server.Documents.Indexes.Index.QueryInternal[TQueryResult](TQueryResult resultToFill, IndexQueryServerSide query, QueryOperationContext queryContext, Boolean pulseDocsReadingTransaction, OperationCancelToken token) in C:\Users\omer.ratsaby\Desktop\RDBQA-64_07082024\60\src\Raven.Server\Documents\Indexes\Index.cs:line 3469
   at Raven.Server.Documents.Indexes.Index.QueryInternal[TQueryResult](TQueryResult resultToFill, IndexQueryServerSide query, QueryOperationContext queryContext, Boolean pulseDocsReadingTransaction, OperationCancelToken token) in C:\Users\omer.ratsaby\Desktop\RDBQA-64_07082024\60\src\Raven.Server\Documents\Indexes\Index.cs:line 3515
   at Raven.Server.Documents.Indexes.Index.Query(IndexQueryServerSide query, QueryOperationContext queryContext, OperationCancelToken token) in C:\Users\omer.ratsaby\Desktop\RDBQA-64_07082024\60\src\Raven.Server\Documents\Indexes\Index.cs:line 3295
   at Raven.Server.Documents.Queries.StaticIndexQueryRunner.ExecuteQuery(IndexQueryServerSide query, QueryOperationContext queryContext, Nullable`1 existingResultEtag, OperationCancelToken token) in C:\Users\omer.ratsaby\Desktop\RDBQA-64_07082024\60\src\Raven.Server\Documents\Queries\StaticIndexQueryRunner.cs:line 38
   at Raven.Server.Documents.Queries.QueryRunner.ExecuteQuery(IndexQueryServerSide query, QueryOperationContext queryContext, Nullable`1 existingResultEtag, OperationCancelToken token) in C:\Users\omer.ratsaby\Desktop\RDBQA-64_07082024\60\src\Raven.Server\Documents\Queries\QueryRunner.cs:line 68
   at Raven.Server.Documents.Handlers.Processors.Queries.DatabaseQueriesHandlerProcessorForGet.GetQueryResultsAsync(IndexQueryServerSide query, QueryOperationContext queryContext, Nullable`1 existingResultEtag, Boolean metadataOnly, OperationCancelToken token) in C:\Users\omer.ratsaby\Desktop\RDBQA-64_07082024\60\src\Raven.Server\Documents\Handlers\Processors\Queries\DatabaseQueriesHandlerProcessorForGet.cs:line 52
   at Raven.Server.Documents.Handlers.Processors.Queries.AbstractQueriesHandlerProcessorForGet`4.ExecuteAsync() in C:\Users\omer.ratsaby\Desktop\RDBQA-64_07082024\60\src\Raven.Server\Documents\Handlers\Processors\Queries\AbstractQueriesHandlerProcessorForGet.cs:line 149
   at Raven.Server.Documents.Handlers.QueriesHandler.Post() in C:\Users\omer.ratsaby\Desktop\RDBQA-64_07082024\60\src\Raven.Server\Documents\Handlers\QueriesHandler.cs:line 14
   at Raven.Server.Routing.RequestRouter.HandlePath(RequestHandlerContext reqCtx) in C:\Users\omer.ratsaby\Desktop\RDBQA-64_07082024\60\src\Raven.Server\Routing\RequestRouter.cs:line 413
   at Raven.Server.RavenServerStartup.RequestHandler(HttpContext context) in C:\Users\omer.ratsaby\Desktop\RDBQA-64_07082024\60\src\Raven.Server\RavenServerStartup.cs:line 192
```

Reply:
Yes, I will add test, I had the fix in my working branch so I pushed it here.